mongo-utils/src/main/java/org/wso2/mongo/actions/AbstractMongoDBAction.java

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.wso2.mongo.actions;
+
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.UpdateResult;
+import org.ballerinalang.jvm.JSONParser;
+import org.ballerinalang.jvm.values.StreamingJsonValue;
+import org.bson.Document;
+import org.wso2.mongo.MongoDBDataSource;
+
+/**
+ * {@code AbstractMongoDBAction} is the base class for all MongoDB actions.
+ */
+
+public class AbstractMongoDBAction {
```

Comment:
The name `AbstractMongoDBAction` but class is not abstract. Shall we make it `abstract`?

Reply:
Fixed